moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_env_bullet.h

```diff
@@ -140,5 +141,8 @@ class CollisionEnvBullet : public CollisionEnv
   void notifyObjectChange(const ObjectConstPtr& obj, World::Action action);
 
   World::ObserverHandle observer_handle_;
+
+  // Lock the collision environment, for thread safety
+  mutable std::mutex collision_env_mutex_;
```

Comment:
Does this lock need to protect anything beside `manager_` / `manager_CCD_`?
If these are the only things that need to be locked, please say so in this comment.
Locking the entire class does not make a huge difference I guess, but is imprecise.

Reply:
Unfortunately, whatever caused the crash in our application does not occur anymore so this is hard for me to test.

I don't know the Bullet setup well enough to guess if it is necessary to protect anything else, so I guess we should leave it locking the whole class.

Reply:
Of course locking only the `contactTest` will fail, because the methods also call `manager_->add/removeCollisionObject`, among others.
Did you include these calls in the same lock?

My message above was mainly about the comment.
There is only one other local class member (`observer_handle_`) and it's not locked by your request.
I'll push a commit to change the comment accordingly.